src/main/java/org/opensearch/neuralsearch/processor/factory/RerankProcessorFactory.java

```diff
@@ -51,22 +58,38 @@ public SearchResponseProcessor create(
     ) {
         RerankType type = findRerankType(config);
         boolean includeQueryContextFetcher = ContextFetcherFactory.shouldIncludeQueryContextFetcher(type);
-        List<ContextSourceFetcher> contextFetchers = ContextFetcherFactory.createFetchers(
-            config,
-            includeQueryContextFetcher,
-            tag,
-            clusterService
-        );
+
+        List<ContextSourceFetcher> contextFetchers = processorRequiresContext(type)
```

Comment:
Can you add a comment above explaining about why this condition is added.

Reply:
Understood, I added this comment 
```
        // Currently the createFetchers method requires that you provide a context map, this branch makes sure we can ignore this on
        // processors that don't need the context map
```